AsyncImageView/Renderers/MulticastedRenderer.swift

```diff
@@ -38,39 +35,32 @@ public final class MulticastedRenderer<
 	}
 
 	public func renderImageWithData(data: Data) -> SignalProducer<ImageResult, NoError> {
-		let property = getPropertyForData(data)
-
-		return property.producer
-			.filter { $0 != nil } // Skip initial `nil` value.
-			.map { $0! }
-			.take(1)
- 	}
-
-	private func getPropertyForData(data: Data) -> ImageProperty {
-		var result: ImageProperty!
+		var result: SignalProducer<ImageResult, NoError>!
 
 		self.cache.modify { cache in
 			var mutableCache = cache
 
-			if let property = cache[data] {
-				result = property
+			if let signal = cache[data] {
+				result = signal
 				return cache
 			}
 
-			result = ImageProperty(
-				initialValue: nil,
-				producer: renderer.createProducerForRenderingData(data)
-					.map(Optional.init)
-			)
+			result = self.createSignalForData(data)
```

Comment:
Whoa that is really cool, now what any client of the cache gets is a consistent view of the progress of the request to get the image, with no extra work. Nice